src/main/asciidoc/java/index.adoc

```diff
@@ -0,0 +1,741 @@
+= Vert.x Codegen
+
+Vert.x Codegen is an annotation processing tool for processing Vert.x API and create API in different JVM lauguages.
+
+Vert.x polyglot langs use code generation for creating https://en.wikipedia.org/wiki/Shim_(computing)[shim] for APIs,
+that is thin wrappers in front of the API type.
+
+A shim is the adaptation of a Vert.x API in a JVM language. A shim uses code generation to generate
+the shim in its language from a Vert.x API, code generation is based on Java annotation processing.
+
+== Vert.x API
+
+A Vert.x API is a collection of Java types annotated with Codegen annotations that respect a set of constraints to
+ensure easy portability of the API to most languages and enable an asynchronous programming model. Such collection
+of API types are grouped in modules.
+
+=== Modules
+
+Modules are declared by annotating a Java package with a `link:../../apidocs/io/vertx/codegen/annotations/ModuleGen.html[@ModuleGen]`
+annotation.
+
+[source,java]
+----
+ModuleGen(name = "acme", groupPackage="com.acme")
+package com.acme.myservice;
+
+import io.vertx.codegen.annotations.ModuleGen;
+----
+
+The module _name_ is a namespace for shims that don't support package like naming, like JavaScript `acme-js`
+or Ruby `acme`, whereas the _group package_ is determines the created package, for instance
+for Groovy `com.acme.groovy.myservice`. The group package must be a prefix of the annotated module.
+
+Vert.x modules uses the reserved name _vertx-XYZ_ and the reserved group package _io.vertx_.
+
```

Comment:
it's unclear what is a vert.x modules. vert.x official modules ? Modules form the official vert.x stack ?